lib/notification.js

```diff
@@ -88,7 +88,7 @@ const NotificationContent = styled.div`
   margin-right: var(--space-1);
 `;
 
-export const Notification = ({ message, live, children, variant, timeout, onClose, ...props }) => {
+export const Notification = ({ message, live, children, variant, timeout, onClose, persistent, ...props }) => {
```

Comment:
do you think the `persistent` prop collides with the `timeout` prop? for example:
```javascript
<Notification timeout={5000} persistent>
  ...
</Notification>
```
looking through the code it seems like the example will be persistent, but is that clear from where we are using the component? we could avoid this by saying `timeout={null | undefined | 0}` means persistent, but i'm not sure that's very clear either!

Reply:
yeah this is something I struggled with - the previous definition has timeout = 0 as persistent, but that didn't feel great to me. I can add a custom function to the `persistent` PropType which throws an error if `persistent === true` and `timeout > 0`; what do you think?

Reply:
do you think a notification would ever be conditionally persistent? if we threw an error, then something like `<Notification persistent={shouldBePersistent} timeout={5000} />` would need to be `<Notification persistent={shouldBePersistent} timeout={shouldBePersistent === false ? 5000 : 0} />` so that we don't pass in something invalid.

maybe a wild idea: what if it was something like `timeout="never"` to disable the timeout?